packages/aws-cdk-lib/core/lib/private/refs.ts

```diff
@@ -19,6 +20,8 @@ import { Stack } from '../stack';
 import { Token, Tokenization } from '../token';
 import { ResolutionTypeHint } from '../type-hints';
 
+const STRING_LIST_REFERENCE_DELIMITER = '||';
```

Comment:
It might be worth exporting/moving the existing constant from the same package instead of duplicating it:

https://github.com/aws/aws-cdk/blob/d4f6946dbc97161749a33d3e89514a8f7ca4aa0a/packages/aws-cdk-lib/core/lib/stack.ts#L41

Reply:
Ack, going to export from here and import into stack because otherwise we'd be making it part of the public api technically